### ISSUE TITLE
Closing LUA Editor causes an assert and disconnects the Asset Processor

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#9612]
**Describe the bug**
Closing the LUA Editor results in the following assert being displayed in the Console and in the popup dialog:

**File:** 
`D:\Git\o3de\Code\Framework\AzCore\AzCore/std/parallel/thread.h`

**Condition:** 
```
'class AZStd::thread &__cdecl AZStd::thread::operator =(class AZStd::thread &&)'
You must call detach or join before you delete/move over the current thread!
```

**Line:** `99`

Additionally, after LUA Editor is closed, the Asset Processor disconnects and needs to be restarted in order to regain its functionality.

Please refer to the attached video for more details.

**Steps to reproduce**
1. Open any level.
2. Launch LUA Editor.
3. Close LUA Editor.

**Expected behavior**
No Asserts are displayed in the Console. The Asset Processor does not disconnect.

**Actual behavior**
An Assert is displayed in the Console. The Asset Processor disconnects.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/168996798-a94a8bde-2530-459a-9f78-130d8b03fb50.mp4

**Found in Branch**
Development (b7fa50e)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 SUPER
 - Memory 16GB

**Attachments**
[Assert.txt](https://github.com/o3de/o3de/files/8714624/Assert.txt)

**Comment by monroegm** [comment]
We believe this is cause by the recent networking changes so we are rerouting to SIG-Network

**Comment by robhaco** [comment]
From initial testing, this is definitely reproduceable in development.  There is an assert that occurs after closing LuaIDE and a modal window that opens which repeats the line seen in log "You must call detach or join before you delete/move over the current thread!"  Once you accept/close this window the AP shuts down.  Interestingly, once the Editor recovers (prompts you to launch the AP again) this no longer happens and the user can open/close LuaIDE freely.  Going to do some additional testing to gather a callstack if possible, etc.

**Comment by AMZN-puvvadar** [comment]
Fix proposed in https://github.com/o3de/o3de/pull/9723

**Comment by AMZN-puvvadar** [comment]
Fix from #9723 merged to development.

**Comment by LB-TomaszSerwanski** [comment]
Fix verified in Development (c221c84).
